ycmd.el

```diff
@@ -2077,61 +2081,18 @@ This produces output for empty alists that ycmd expects."
 (define-hmac-function ycmd--hmac-function
   ycmd--secure-hash 64 64)
 
-(defun ycmd--options-contents (hmac-secret)
-  "Return a struct with ycmd options and the HMAC-SECRET applied.
-The struct can be json encoded into a file to create a ycmd
-options file.
-
-When we start a new ycmd server, it needs an options file.  It
-reads this file and then deletes it since it contains a secret
-key.  So we need to generate a new options file for each ycmd
-instance.  This function effectively produces the contents of that
-file."
-  (let ((hmac-secret (base64-encode-string hmac-secret))
-        (global-config (or ycmd-global-config ""))
-        (extra-conf-whitelist (or ycmd-extra-conf-whitelist []))
-        (confirm-extra-conf (if (eq ycmd-extra-conf-handler 'load) 0 1))
```

Comment:
`ycmd-extra-conf-handler` is used also in `ycmd--handle-extra-conf-exception` so it needs to be saved from the json.

Reply:
> `ycmd-extra-conf-handler` is used also in `ycmd--handle-extra-conf-exception` so it needs to be saved from the json.

I am not sure about that, because that is just how emacs-ycmd displays exceptions from ycmd. ycmd's option to confirm config config, is a little bit different, it is about does the server send this exceptions or not.

Reply:
When you set `confirm_extra_conf` to `1` in the json file, you will get an extra conf exception when ycmd tries to load an extra conf file. When `ycmd-extra-conf-handler` is set to load at the same time it will load the extra conf without the client having to confirm.

Reply:
I think `confirm_extra_conf` and `ycmd-extra-conf-handler` need to be synced somehow.

Reply:
I do not have a strong opinion, but I find it ok to keep `ycmd-extra-conf-handler` even, though it does overlap in behavior with the ycmd's option, it still has some neat behavior control over how emacs handles the exceptions, that someone might like. Ofc, it is not useful if the exceptions are totally disabled from the JSON settings, but I think it is fine.